src/ceph-volume/ceph_volume/devices/raw/activate.py

```diff
@@ -44,16 +45,49 @@ def activate_bluestore(meta, tmpfs, systemd):
     # correctly every time
     process.run(['ln', '-snf', meta['device'], os.path.join(osd_path, 'block')])
     system.chown(os.path.join(osd_path, 'block'))
-    system.chown(osd_path)
 
-#    if systemd:
-        # write me
-        # enable the OSD
-        #systemctl.enable_osd(osd_id)
+    wal_device = os.getenv("CEPH_VOLUME_OSD_WAL_DEVICE")
+    if wal_device:
+        out, err, ret = process.call(['ceph-bluestore-tool', 'show-label', '--dev', wal_device])
+        if err:
+            terminal.error('ceph-bluestore-tool failed to run. %s'% err)
+            raise SystemExit(1)
+        if ret:
+            terminal.error('no label on %s'% wal_device)
+            raise SystemExit(1)
+        oj = json.loads(''.join(out))
+        if wal_device not in oj:
+            terminal.error('%s not in the output of ceph-bluestore-tool, buggy?'% wal_device)
+            raise SystemExit(1)
```

Comment:
This should probably be put into a `util/prepare.py`, would make for easier testing too.